_posts/events-magic.md

```diff
@@ -0,0 +1,81 @@
+
+That means that swapping the communication protocol for something that, on paper, promises less or no coupling is the solution to all the coupling-related issues.
+
+Let me use a couple of examples. Two components, A and B, communicate by storing data in a database table. A writes some data, which B then reads to continue the business process.
+
+Oh! They use the same storage, and if A and B were (micro)services, that's bad. They are coupled with the same technology and many other irrelevant (pardon my candor) things. The decision is to remove that coupling by, e.g., replacing the communication protocol with HTTP, gRPC, some obscure JSON-based stuff, messages on a queue, or events on a broker.
+
+The problem is that coupling doesn't go anywhere, and things might worsen. What was a monolith with some coupling could transform into a distributed monolith with coupling in the same places, creating many new problems.
+
+## Where should we focus instead?
+
+Events should be the natural consequence of something else, not the primary goal. We should focus on [autonomous components and service boundaries](https://milestone.topics.it/2023/05/17/back-to-basics-boundaries.html). Finding service boundaries is the first step. That alone enables the defining of autonomous components. At this point, we'll realize that the natural way for autonomous components to communicate, especially across service boundaries, is by using thin events carrying only identifiers and no other data.
+
+If we start from the end by introducing events first when the service boundaries are ill-defined, we're destined to face the dreaded "how do I get the data stored in that other service that I need here now to perform my task" question. There can only be three answers:
+
+- Let's expose a RESTful API to allow querying for the needed data.
+- Let's share the needed data using the events and messages we're already sending or publishing.
+- Maybe our service boundaries are wrong.
+
+It's tough to accept the truth presented by the third answer, and it's much easier to go with one of the other two, with a predominance for the first.
+
+Unfortunately, not recognizing that the boundaries are wrong leads to designing a distributed monolith affected by schema and temporal coupling to begin with.
+
+It's also crucial to understand that It's not a monolithic process. We don't need to sort out all boundaries perfectly. Then, define all the autonomous components. And finally, determine who publishes what and who subscribes. It's a messy process that leads to rough boundaries or isolates one or more services from the rest of the system. By looking at the relationships between the identified services and components, it's possible to start crafting some events. We're ready to return to the drawing board and address another portion of the system. The other critical part is that we should not fear any rework of previously identified boundaries and components, which might lead to reshaping events.
```

Comment:
"It's not a monolithic process" maybe not "ridig" or "straightforward"?

I'd go even further and say it's impossible to get it right from the beginning and that's the essence of the concept of domain crunching IMO :D